traffic_ops/traffic_ops_golang/servers.go

```diff
@@ -0,0 +1,253 @@
+
+		q := r.URL.Query()
+		resp, err := getServersResponse(q, db, privLevel)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		respBts, err := json.Marshal(resp)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, "%s", respBts)
+	}
+}
+
+func getServersResponse(q url.Values, db *sql.DB, privLevel int) (*tcstructs.ServersResponse, error) {
+	servers, err := getServers(q, db, privLevel)
+	if err != nil {
+		return nil, fmt.Errorf("getting servers response: %v", err)
+	}
+
+	resp := tcstructs.ServersResponse{
+		Response: servers,
+	}
+	return &resp, nil
+}
+
+func getServers(q url.Values, db *sql.DB, privLevel int) ([]tcstructs.Server, error) {
+
+	rows, err := db.Query(selectServersQuery())
+
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	servers := []tcstructs.Server{}
+
+	const HiddenField = "********"
+
+	for rows.Next() {
+		var cachegroup sql.NullString
+		var cachegroupId sql.NullInt64
+		var cdnId sql.NullInt64
+		var cdnName sql.NullString
+		var domainName sql.NullString
+		var guid sql.NullString
+		var hostName sql.NullString
+		var httpsPort sql.NullInt64
+		var id sql.NullInt64
+		var iloIpAddress sql.NullString
+		var iloIpGateway sql.NullString
+		var iloIpNetmask sql.NullString
+		var iloPassword sql.NullString
+		var iloUsername sql.NullString
+		var interfaceMtu sql.NullInt64
+		var interfaceName sql.NullString
+		var ip6Address sql.NullString
+		var ip6Gateway sql.NullString
+		var ipAddress sql.NullString
+		var ipGateway sql.NullString
+		var ipNetmask sql.NullString
+		var lastUpdated sql.NullString
+		var mgmtIpAddress sql.NullString
+		var mgmtIpGateway sql.NullString
+		var mgmtIpNetmask sql.NullString
+		var offlineReason sql.NullString
+		var physLocation sql.NullString
+		var physLocationId sql.NullInt64
+		var profile sql.NullString
+		var profileDesc sql.NullString
+		var profileId sql.NullInt64
+		var rack sql.NullString
+		var revalPending sql.NullBool
+		var routerHostName sql.NullString
+		var routerPortName sql.NullString
+		var status sql.NullString
+		var statusId sql.NullInt64
+		var tcpPort sql.NullInt64
+		var serverType sql.NullString
+		var serverTypeId sql.NullInt64
+		var updPending sql.NullBool
+		var xmppId sql.NullString
+		var xmppPasswd sql.NullString
+		if err := rows.Scan(&cachegroup, &cachegroupId, &cdnId, &cdnName, &domainName, &guid, &hostName, &httpsPort, &id, &iloIpAddress, &iloIpGateway, &iloIpNetmask, &iloPassword, &iloUsername, &interfaceMtu, &interfaceName, &ip6Address, &ip6Gateway, &ipAddress, &ipGateway, &ipNetmask, &lastUpdated, &mgmtIpAddress, &mgmtIpGateway, &mgmtIpNetmask, &offlineReason, &physLocation, &physLocationId, &profile, &profileDesc, &profileId, &rack, &revalPending, &routerHostName, &routerPortName, &status, &statusId, &tcpPort, &serverType, &serverTypeId, &updPending, &xmppId, &xmppPasswd); err != nil {
```

Comment:
Why not just scan directly into the new object? This way is useful if you've got nullable columns that need to be handled carefully, but you've already taken care of that with `coalesce`.